analyzers/src/SonarAnalyzer.Common/Rules/PropertiesAccessCorrectFieldBase.cs

```diff
@@ -251,6 +260,13 @@ public IFieldSymbol GetSingleMatchingFieldOrNull(IPropertySymbol propertySymbol)
 
             private static bool AreCanonicalNamesEqual(string name1, string name2) =>
                 name1.Equals(name2, StringComparison.OrdinalIgnoreCase);
+
+            private bool FieldMatchesTheProperty(IFieldSymbol field, IPropertySymbol property) =>
+                // We're not caching the property name as only expect to be called once per property
+                AreCanonicalNamesEqual(fieldToStandardNameMap[field], GetCanonicalFieldName(property.Name))
+                && field.Type.Equals(property.Type)
+                && ((property.IsStatic && field.IsStatic) || (!property.IsStatic && !field.IsStatic))
+                && !field.IsConst;
```

Comment:
It might be worth rearranging the order of the checks. For example it should be faster to check if the field is not a constant than to do the string comparisions and "Canonification" of the name.